sources/common/infoschema.go

```diff
@@ -133,6 +139,141 @@ func ProcessData(conv *internal.Conv, infoSchema InfoSchema) {
 	}
 }
 
+func ProcessDataWithDataproc(conv *internal.Conv, infoSchema InfoSchema, dataprocConfig map[string]string) error {
+
+	orderTableNamesByID := ddl.GetSortedTableIdsBySpName(conv.SpSchema)
+	numberOfTables := int64(len(orderTableNamesByID))
+
+	if !conv.Audit.DryRun {
+		conv.Audit.Progress = *internal.NewProgress(numberOfTables, "Writing data to Spanner via Dataproc", internal.Verbose(), false, int(internal.DataWriteInProgress))
+	}
+
+	progressCtr := 0
+
+	for _, spannerTableID := range orderTableNamesByID {
+
+		srcTable := conv.SrcSchema[spannerTableID].Name
+
+		srcSchema := conv.SrcSchema[spannerTableID]
+
+		primaryKeys, _, _ := infoSchema.GetConstraints(conv, SchemaAndName{Name: srcTable, Schema: srcSchema.Schema})
+
+		id, err := TriggerDataprocTemplate(srcTable, srcSchema.Schema, strings.Join(primaryKeys, ","), dataprocConfig)
+		if err != nil {
+			return err
+		}
+		if conv.DataFlush != nil {
+			conv.DataFlush()
+		}
+
+		if !conv.Audit.DryRun {
+			progressCtr++
+			conv.Audit.Progress.MaybeReport(int64(progressCtr))
+		}
+
+		//TODO: eenclona@ will remove hardcoded us-central1 to be parametarized
+		url := fmt.Sprintf("https://pantheon.corp.google.com/dataproc/batches/us-central1/%s", id)
```

Comment:
This an internal url, will most likely need to be replaced by a `googleapis` link in addition to parameterization

Reply:
Done, 
Commit: https://github.com/cloudspannerecosystem/harbourbridge/pull/520/commits/9012732e7958d82f494e148d78387f8c8c7eb447